### DISCUSSION TITLE
Understanding CatalogsClient

### DISCUSSION THREAD

**Discussion by karoldeland** [mehdihadeli/vertical-slice-api-template#27] [Q&A]
Hi @mehdihadeli, 
Great work in this template. I believe it's the better template out there. If possible, I would like to understand the benefit of having the CatalogApiClient in the Vertical.Slice.Template project. If I understand correctly, it's only used by tests and it's a client class targeting the product API in the same assembly. For me, it feels like a circular dependency. Could you explain a little?

Thank you for you work.

**Comment by karoldeland** [comment]
Would be great if you can explain the purpose of the Vertical.Slice.Template.ApiClient project too.

**Comment by mehdihadeli** [comment]
Hi,
Thanks for your feedback.
I will refactor this template and add some new features soon :)
Regarding `APIClient` this is just a sample for some use cases in microservices world, for example we have a customer microservice and we need to call catalogs api endpoints on top of [catalog api address](https://github.com/mehdihadeli/vertical-slice-api-template/blob/main/src/Vertical.Slice.Template.ApiClient/Vertical.Slice.Template.ApiClient.csproj#L23), for doing this we should configure catalog address in our [Client csproj](https://github.com/mehdihadeli/vertical-slice-api-template/blob/main/src/Vertical.Slice.Template.ApiClient), It will create all needed code for calling catalog micro. but this client project is not mandatory and I should create some flag like `dotnet new generate-api-client --source=http://localhost:4000/swagger/v1/swagger.json --name="CatalogsApiClient"` for generating it optionally.
